### PR TITLE
devices: Add Environmental.SenseContinuous(); implement for bme280 

### PR DESCRIPTION
**Pull request by maruel**
- Unexport the Standby parameter.
- Have SenseContinuous accept time.Duration period instead.
- Set the filter only when using SenseContinuous.
- Do not enable automatic sensing upon initialization.
- Use forced mode by default.

### CODE REVIEW

**Comment by mattetti** [comment]
smoke tests are failing because they weren't updated:

> devices/bme280/bme280smoketest/bme280smoketest.go:127: unknown field 'Standby' in struct literal of type bme280.Opts

File: devices/bme280/bme280.go [hunk 1]
```diff
@@ -31,113 +34,209 @@ import (
...
+		0xF4, byte(d.opts.Temperature)<<5 | byte(d.opts.Pressure)<<2 | byte(normal),
+	})
+	if err != nil {
+		return nil, err
+	}
+	sensing := make(chan devices.Environment)
+	d.stop = make(chan struct{})
+	go func() {
+		defer close(sensing)
+		d.senseContinuous(interval, sensing, d.stop)
```
Comment (mattetti) [inline devices/bme280/bme280.go]: nit: I got confused not noticing that we were calling a different method and not making a recursive call. The recursive call didn't make a lot of sense (pun intended) and the args were different. A suggestions would be to call the unexported methods `sensingContinuously`.

File: devices/bme280/bme280.go [hunk 2]
```diff
@@ -31,113 +34,209 @@ import (
...
-	// 4.1.
-	// Pressure: 0xF7~0xF9
-	// Temperature: 0xFA~0xFC
-	// Humidity: 0xFD~0xFE
-	buf := [0xFF - 0xF7]byte{}
-	if err := d.readReg(0xF7, buf[:]); err != nil {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+	if d.stop != nil {
+		return errors.New("bme280: already sensing continuously")
```
Comment (mattetti) [inline devices/bme280/bme280.go]: My bad, the lock seems to be released when running continuously in the background. I'm now kind of confused about the use case of the mutex. I thought it was a way to protect the user from running multiple polls in parallel.

File: devices/bme280/bme280.go [hunk 3]
```diff
@@ -31,113 +34,209 @@ import (
...
-	F16  Filter = 4
+	NoFilter Filter = 0
+	F2       Filter = 1
+	F4       Filter = 2
+	F8       Filter = 3
+	F16      Filter = 4
 )
 
-// Dev is an handle to a bme280.
+// Dev is an handle to an initialized bme280.
```
Comment (maruel) [inline devices/bme280/bme280.go]: Done

File: devices/bme280/bme280.go [hunk 4]
```diff
@@ -31,113 +34,209 @@ import (
...
+	mu   sync.Mutex
+	stop chan struct{}
 }
 
 func (d *Dev) String() string {
 	return fmt.Sprintf("BME280{%s}", d.d)
 }
 
-// Sense returns measurements as °C, kPa and % of relative humidity.
+// Sense request a single one time measurement as °C, kPa and % of relative
```
Comment (maruel) [inline devices/bme280/bme280.go]: Done

File: devices/bme280/bme280.go (hunk 2 above)
Comment (maruel) [inline devices/bme280/bme280.go]: The lock is only held while talking to the device. It serves two purpose:
- serialize access to the device
- serialize access to `d.stop`

File: devices/bme280/bme280.go (hunk 1 above)
Comment (maruel) [inline devices/bme280/bme280.go]: Done

**Comment by maruel** [comment]
Paging @quinte17 since this PR will fix issue #20.

**Comment by mattetti** [comment]
+1

**Pull request merged**